src/QsCompiler/LlvmBindings/BitcodeModule.cs

```diff
@@ -120,19 +118,60 @@ private BitcodeModule(LLVMModuleRef handle)
             }
         }
 
-        /// <summary>Gets the <see cref="DebugInfoBuilder"/> used to create debug information for this module</summary>
-        /// <remarks>The builder returned from this property is lazy constructed on first access so doesn't consume resources unless used.</remarks>
+        /// <summary>Creates and returns a new <see cref="DebugInfoBuilder"/> used to create debug information for this module</summary>
+        public DebugInfoBuilder CreateDIBuilder()
+        {
+            this.ThrowIfDisposed();
```

Comment:
Was `this.ThrowIfDisposed()` taken from the Llvm.NET repo?

Reply:
No, I added it in when I created this function because it's used throughout our BitcodeModule.cs